sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -969,6 +969,12 @@ class GpuRegExpExtractMeta(
   override def tagExprForGpu(): Unit = {
     GpuRegExpUtils.tagForRegExpEnabled(this)
 
+    ShimLoader.getShimVersion match {
```

Comment:
Do we have/want a follow on issue to understand this? Did we test it on opensource deltalake to know if we have the same problem?

Reply:
I filed #5055 as a followup to investigate.  I did not test with open source Delta Lake, mostly because I was not able to reproduce the problem by simply writing a Delta Lake file and trying to read it.  @viadea tested this patch on a customer use-case, but I was not able to reproduce it myself.

I did check the [Delta Lake source](https://github.com/delta-io/delta) and did not see it generating `RegExpExtract`/`regexp_extract` usage.  It did have references to `RegExpExtract` but only in the context of a large list of expressions that can be part of a generated column, and this list is only used as a sanity-check on columns to support or it throws.